pkg/azurefile/controllerserver.go

```diff
@@ -924,7 +924,11 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 
 	klog.V(2).Infof("created share snapshot: %s, time: %v, quota: %dGiB", itemSnapshot, itemSnapshotTime, itemSnapshotQuota)
 	if itemSnapshotQuota == 0 {
-		itemSnapshotQuota = defaultAzureFileQuota
+		fileshare, err := d.cloud.FileClient.WithSubscriptionID(subsID).GetFileShare(ctx, rgName, accountName, fileShareName, "")
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to get file share(%s) quota: %v", fileShareName, err)
+		}
+		itemSnapshotQuota = *fileshare.ShareQuota
```

Comment:
use `ptr.Deref(fileshare.ShareQuota, 0)` to avoid nil pointer panic

Reply:
I think here should be `itemSnapshotQuota = pointer.Int32Deref(fileshare.ShareQuota, defaultAzureFileQuota)` since 0 will also return an error